StitchCore/Sources/StitchCore/StitchClientConfiguration.swift

```diff
@@ -34,6 +41,11 @@ public struct StitchClientConfigurationImpl: StitchClientConfiguration, Buildee
      */
     public let baseURL: String
 
+    /**
+     * The local directory in which Stitch can store any data (e.g. embedded MongoDB data directory).
+     */
+    public let dataDirectory: URL?
```

Comment:
[q] Why is the optional? It seems like we can't really function without this.

Reply:
In the Android SDK it can be null, so kept it that way in iOS for consistency's sake. 

For now it's really only necessary when running embedded MongoDB, and for the server versions of the SDK we may make it fine not to configure it by default. I'm happy to make it non-optional though if we make the same change to Android. Thoughts @edaniels ?

Reply:
Should just make it non opt and validate